pkg/workceptor/kubernetes.go

```diff
@@ -103,12 +106,40 @@ func (kw *kubeUnit) createPod(env map[string]string) error {
 	if err != nil {
 		return err
 	}
-	pod := &corev1.Pod{
-		ObjectMeta: metav1.ObjectMeta{
-			GenerateName: kw.namePrefix,
-			Namespace:    ked.KubeNamespace,
-		},
-		Spec: corev1.PodSpec{
+	spec := corev1.PodSpec{}
+	if ked.KubePodSpec != "" {
+		reader := bytes.NewReader([]byte(ked.KubePodSpec))
+		decoder := yaml.NewYAMLOrJSONDecoder(reader, 1024)
+		err := decoder.Decode(&spec)
+		if err != nil {
+			return err
+		}
+		foundWorker := false
+		for _, container := range spec.Containers {
+			if container.Name == "worker" {
+				if !container.Stdin {
```

Comment:
There's no need to check this before setting it.  This is just a boolean variable - there ere are no side effects from setting it.